vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_controller_standard.go

```diff
@@ -231,6 +231,27 @@ func (as *availabilitySet) UpdateVM(ctx context.Context, nodeName types.NodeName
 	return nil
 }
 
+// UpdateVMAsync updates a vm asynchronously
+func (as *availabilitySet) UpdateVMAsync(ctx context.Context, nodeName types.NodeName) (*azure.Future, error) {
+	vmName := mapNodeNameToVMName(nodeName)
+	nodeResourceGroup, err := as.GetNodeResourceGroup(vmName)
+	if err != nil {
+		return nil, err
+	}
+	klog.V(2).Infof("azureDisk - update(%s): vm(%s)", nodeResourceGroup, vmName)
+	// Invalidate the cache right after updating
+	defer func() {
+		_ = as.DeleteCacheForNode(vmName)
+	}()
+
+	future, rerr := as.VirtualMachinesClient.UpdateAsync(ctx, nodeResourceGroup, vmName, compute.VirtualMachineUpdate{}, "update_vm")
+	klog.V(2).Infof("azureDisk - update(%s): vm(%s) - returned with %v", nodeResourceGroup, vmName, rerr)
+	if rerr != nil {
+		return future, rerr.Error()
```

Comment:
should we be returning nil instead of ` future` here?

Reply:
This seems to be inconsistent from the way the rest of the code.

Reply:
Although we were pretty sure that we wouldn't be getting an error with HTTP status code of 2xx for `UpdateAsync` calls, we just wanted to be safe and have a comprehensive way of capturing `PartialUpdateErrors` for these methods.